geo/src/algorithm/coords_iter.rs

```diff
@@ -278,6 +280,35 @@ impl<'a, T: CoordinateType> Iterator for GeometryCoordsIter<'a, T> {
     }
 }
 
+impl<'a, T: CoordinateType> fmt::Debug for GeometryCoordsIter<'a, T> {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
I think I'd expect something akin to [`impl Debug for Vec::Iter`](https://doc.rust-lang.org/src/core/slice/iter.rs.html#73) which includes details about the vec contents, which is more likely what you're interested in rather than just the type.

```
fn main() {
    let v = vec!["a", "b", "c"];
    println!("v.iter() debug: {:?}", v.iter());
    println!("v debug: {:?}", v);
}
```

Which outputs:
```
v.iter() debug: Iter(["a", "b", "c"])
v debug: ["a", "b", "c"]
```

Reply:
Thanks for the tip... that does look better, more standard.

The fix was simple for everything but the "GeometryCollection" item .. I am having touble dealing with the match type (i)

which in this case is of type

&Box< dyn Iterator<item...>>

I played with it and could not find a good way of dynamically casting into something that would compile.

Any advice would be welcome.

Reply:
@martinfrances107 what's code have you tried, and what error do you get?

Reply:
> what's the code you have tried

inside the implmentation of "fmt::Debug for GeometryCoordsIter<'a, T>"

            GeometryCoordsIter::GeometryCollection(i) => {
                fmt.debug_tuple("GeometryCollection").field(i).finish()
            }

I will post the error below ... 

speaking conversationally .. I think maybe I need to alter a definition in geometry_collection.rs 
enforcing the type contract  ( "+ std::fmt::Debug" ) I just can't seem to get a working version.

or maybe I need to implement Debug for the GeometryCoordsIter .. and just have a simple loop which reports the debug info for each element ?

That is where I am stuck 
```

`dyn std::iter::Iterator<Item = geo_types::Coordinate<T>>` doesn't implement `std::fmt::Debug`

`dyn std::iter::Iterator<Item = geo_types::Coordinate<T>>` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`

help: the trait `std::fmt::Debug` is not implemented for `dyn std::iter::Iterator<Item = geo_types::Coordinate<T>>`
note: required because of the requirements on the impl of `std::fmt::Debug` for `std::boxed::Box<dyn std::iter::Iterator<Item = geo_types::Coordinate<T>>>`
note: required for the cast to the object type `dyn std::fmt::Debug`


```

Reply:
the `Iterator` itself doesn't implement `Debug`, but each of the items do so we need to loop through the iterator.

does something like this work?

```rust
let debug_tuple = fmt.debug_tuple("GeometryCollection");

for geometry in i {
    debug_tuple.field(i);
}

debug_tuple.finish();
```